src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1040,7 +1040,7 @@ protected internal struct EventData
             /// Address where the one argument lives (if this points to managed memory you must ensure the
             /// managed object is pinned.
             /// </summary>
-            public IntPtr DataPointer { get { return (IntPtr)m_Ptr; } set { m_Ptr = unchecked((long)value); } }
+            public IntPtr DataPointer { get { return (IntPtr)m_Ptr; } set { m_Ptr = unchecked((ulong)value); } }
```

Comment:
❓ Are you sure this fixes the issue? I'm still seeing sign-extension of the underlying value in the C# Interactive window.

```
> IntPtr.Size
4
> var dataPointer = (IntPtr)unchecked((int)0xdeadf00d);
> dataPointer.ToString("x16")
"00000000deadf00d"
> ((ulong)dataPointer).ToString("x16")
"ffffffffdeadf00d"
```

I would have expected to see the following here:

```csharp
unsafe
{
  m_Ptr = (ulong)(void*)value;
}
```